File: api_examples/validate_receipt_smtp.py
""" This script is an example of sending an email via smtp
and confirming its receipt using the mailsac api"""

import time
import sys
import smtplib
import email.utils
import requests
from email.mime.text import MIMEText

"""
Checks if a message from a given address in the a specific mailsac
inbox. If it is it returns when the message was recived, if not it returns
a message stating the message was not received"""
def check_received(receive_address, send_address, base_url, headers):
    api_url = '{0}addresses/{1}/messages'.format(base_url, receive_address)
    response = requests.get(api_url, headers=headers)
    for message in response.json():
        if message['from'][0]['address'] == send_address:
            return message['received']
        return '{0} has not received an email from {1}'.format(receive_address, send_address)


SMTP_USERNAME = 'mysmtp_user'
SMTP_PASSWORD = 'mysmtp_password'
SMTP_SERVER = 'mysmtp_server.company.com'
SMTP_PORT = 587
FROM_ADDRESS = 'myuser@company.com'
FROM_NAME = 'MyTest User'
SUBJECT = "Testing email to mailsac"

API_TOKEN = 'MY_API_TOKEN_FROM_MAILSAC'
BASE_URL = 'https://mailsac.com/api/'

BODY_TEXT = ("Mailsac SMTP Validate Email Send\r\n"
             "This email was sent using the SMTP to test receipt of an email."
            )

for x in range(1, 10):
    try:
        to_address = 'user{}@mailsac.com'.format(x)
        msg = MIMEText(BODY_TEXT)
        msg['Subject'] = SUBJECT
        msg['From'] = email.utils.formataddr((FROM_NAME, FROM_ADDRESS))
        msg['To'] = to_address

        conn = smtplib.SMTP(SMTP_SERVER, SMTP_PORT)
        conn.set_debuglevel(True)
        conn.ehlo()
        conn.starttls()
        conn.ehlo()
        conn.login(SMTP_USERNAME, SMTP_PASSWORD)
        try:
            conn.sendmail(FROM_ADDRESS, to_address, msg.as_string())
            conn.close()
        except Exception as e:
            print("Error: ", e)
        else:
            print("Email Sent!")
    except Exception as ex:
        print(ex)

time.sleep(30)

for x in range(1, 10):
    print(check_received('user{}@mailsac.com'.format(x), send_address=FROM_ADDRESS,
                      base_url=BASE_URL, headers={'Mailsac-Key': API_TOKEN}))

File: api_examples/validate_receipt_smtp.js
const request = require('superagent');
const nodemailer = require('nodemailer');

const checkReceived = (receiveAddress, sendAddress, baseURL, headers) => {
    const apiURL = `${baseURL}/addresses/${receiveAddress}/messages`;
    request
        .get(apiURL)
        .set(headers)
        .then(res => {
            res.body.forEach((message) => {
                if (message.from[0].address == sendAddress) {
                    console.log(message.received)
                } else {
                    console.log(`${receiveAddress} has not received an email from ${sendAddress}`)
                }
            })
        })
        .catch(err => {
            console.log(err.message)
        });
}

const SMTP_USERNAME = 'mysmtp_user';
const SMTP_PASSWORD = 'mysmtp_password';
const SMTP_SERVER = 'mysmtp_server.company.com';
const SMTP_PORT = 587;
const FROM_ADDRESS = 'myuser@company.com';
const FROM_NAME = 'MyTest User';
const SUBJECT = 'Testing email to Mailsac';

const API_TOKEN = 'MY_API_TOKEN_FROM_MAILSAC';
const BASE_URL = 'https://mailsac.com/api';

const BODY_TEXT = "Mailsac SMTP validate Email Send\r\nThis email was sent using the SMTP to test receipt of an email.";

for (let i = 1; i < 11; i++) {
    const transporter = nodemailer.createTransport({
        host: SMTP_SERVER,
        port: SMTP_PORT,
        auth: {
            user: SMTP_USERNAME,
            pass: SMTP_PASSWORD,
        },
    });

    const result = transporter.sendMail({
        from: `"${FROM_NAME}" <${FROM_ADDRESS}>`,
        to: `user${i}@mailsac.com`,
        subject: SUBJECT,
        text: BODY_TEXT,
    }, (err, res) => {
        if (err) {
            console.log(err)
        } else {
            console.log("Email sent!")
        }
    });
}

setTimeout(() => {
    for (let i = 1; i < 11; i++) {
        checkReceived(`user${i}@mailsac.com`, FROM_ADDRESS, BASE_URL, {'Mailsac-Key': API_TOKEN});
    }
}, 30);